python/src/aiconfig/editor/client/src/components/EditorContainer.tsx

```diff
@@ -594,14 +617,53 @@ export default function EditorContainer({
                 output: event.data,
               });
             } else if (event.type === "aiconfig") {
+              dispatch({
+                type: "CONSOLIDATE_AICONFIG",
+                action: {
+                  ...action,
+                  // Ensure we keep the prompt in a running state since this is an in-progress update
+                  isRunning: true,
+                },
+                config: event.data,
+              });
+            } else if (event.type === "aiconfig_complete") {
               dispatch({
                 type: "CONSOLIDATE_AICONFIG",
                 action,
                 config: event.data,
               });
             }
           },
-          enableStreaming
+          (event) => {
+            console.log(
+              `Error running prompt ${promptName}: ${JSON.stringify(event)}`
+            );
+            if (event.type === "error") {
+              //throw new Error(event.data.message);
+
+              if (event.data.code === 499) {
+                // This is a cancellation
+                // Reset the aiconfig to the state before we started running the prompt
```

Comment:
IMO we shouldn't reset the config. If you cancel while streaming it should just stop the output where it is instead of clearing it